x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/persistence/DatafeedConfigProvider.java

```diff
@@ -131,6 +159,44 @@ public void onFailure(Exception e) {
         });
     }
 
+    /**
+     * Find any datafeeds that are used by job {@code jobid} i.e. the
+     * datafeed that references job {@code jobid}.
+     *
+     * In theory there should never be more than one datafeed referencing a
+     * particular job.
+     *
+     * @param jobId     The job to find
+     * @param listener  Datafeed Id listener
+     */
+    public void findDatafeedForJobId(String jobId, ActionListener<Set<String>> listener) {
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder().query(buildDatafeedJobIdQuery(jobId));
+        sourceBuilder.sort(DatafeedConfig.ID.getPreferredName());
+        sourceBuilder.fetchSource(false);
+        sourceBuilder.docValueField(DatafeedConfig.ID.getPreferredName());
+
+        SearchRequest searchRequest = client.prepareSearch(AnomalyDetectorsIndex.configIndexName())
+                .setIndicesOptions(IndicesOptions.lenientExpandOpen())
+                .setSource(sourceBuilder).request();
+
+        executeAsyncWithOrigin(client.threadPool().getThreadContext(), ML_ORIGIN, searchRequest,
+                ActionListener.<SearchResponse>wrap(
```

Comment:
Is the type qualification necessary? Could we just do `ActionListener.wrap`?

Reply:
Yes because client::search is overloaded